dexs/uniswap-v4.ts

```diff
@@ -1,53 +1,315 @@
+//           OR (swaps.chain = 'base' AND swaps.contract_address = 0x498581ff718922c3f8e6a244956af099b2652b2b)
+//           OR (swaps.chain = 'unichain' AND swaps.contract_address = 0x1f98400000000000000000000000000000000004)
+//           OR (swaps.chain = 'optimism' AND swaps.contract_address = 0x9a13f98cb987694c9f086b1f5eb990eea8264ec3)
+//           OR (swaps.chain = 'arbitrum' AND swaps.contract_address = 0x360e68faccca8ca495c1b759fd9eee466db9fb32)
+//           OR (swaps.chain = 'polygon' AND swaps.contract_address = 0x67366782805870060151383f4bbff9dab53e5cd6)
+//           OR (swaps.chain = 'blast' AND swaps.contract_address = 0x1631559198a9e474033433b2958dabc135ab6446)
+//           OR (swaps.chain = 'zora' AND swaps.contract_address = 0x0575338e4c17006ae181b47900a84404247ca30f)
+//           OR (swaps.chain = 'worldchain' AND swaps.contract_address = 0xb1860d529182ac3bc1f51fa2abd56662b7d13f33)
+//           OR (swaps.chain = 'ink' AND swaps.contract_address = 0x360e68faccca8ca495c1b759fd9eee466db9fb32)
+//           OR (swaps.chain = 'avalanche_c' AND swaps.contract_address = 0x06380c0e0912312b5150364b9dc4542ba0dbbc85)
+//           OR (swaps.chain = 'bnb' AND swaps.contract_address = 0x28e2ea090877bf75740558f6bfb36a5ffee9e9df)
+//         )
+//     )
+//     SELECT
+//       chain,
+//       token,
+//       SUM(amount) AS totalSwapAmount,
+//       SUM(feeAmount) AS totalSwapFee
+//     FROM transactions
+//     GROUP BY chain, token
+//   `;
+// }
 
-chains.forEach(chain => adapter.adapter[chain] = { fetch: fetch as any })
+// async function prefetchWithDune(options: FetchOptions) {
+//   return await queryDune('3996608',{
+//     fullQuery: UNISWAP_V4_DUNE_QUERY(options.fromTimestamp, options.toTimestamp),
+//   });
+// }
 
-export default adapter;
+function getPoolKey(poolId: string): string {
+  return poolId.slice(0, 52);
+}
 
-const dataCache = {} as any
+async function fetch(options: FetchOptions) {
+  const dailyFees = options.createBalances()
+  const dailyVolume = options.createBalances()
 
-async function fetch(_: any, _1: any, { api, startOfDay, }: FetchOptions) {
-  switch (api.chain) {
-    case 'unichain': api.chainId = 130; break;
+  const config = Configs[options.chain];
+  if (!config) {
+    throw Error(`config not found for chain ${options.chain}`);
   }
-  const endpoint = `https://interface.gateway.uniswap.org/v2/uniswap.explore.v1.ExploreStatsService/ExploreStats?connect=v1&encoding=json&message=%7B%22chainId%22%3A%22${api.chainId}%22%7D`
 
-  try {
-    if (!dataCache[endpoint]) dataCache[endpoint] = await httpGet(endpoint, {
-      headers: {
-        'origin': 'https://app.uniswap.org',
-      }
-    })
-    const res = await dataCache[endpoint]
-    const datapoint = res.stats.historicalProtocolVolume.Month.v4.find((i: any) => i.timestamp === startOfDay)
+  let events: Array<any> = [];
+
+  if (config.source === 'LOGS') {
```

Comment:
no need to split between logs and indexer, if indexer is supported, sdk automatically pulls from indexer, else pulls logs from chain rpc